1-Draft/RFCXXXX-Umask-and-Ulimit.md

```diff
@@ -0,0 +1,276 @@
+---
+RFC: XXXX
+Author: Jim Truher
+Status: Draft
+Area: Strict Mode
+Comments Due: 7/1/2019
+---
+
+# Umask and Ulimit
+
+Unix shells have a number of built in commands which are used to configure the shell's behavior.
+This RFC proposes two new areas of control for PowerShell, namely, `umask` and `ulimit`
+
+## UMASK
+
+On Linux and MacOS systems when a file is created, it is created with a specific set of permissions.
+This is known as _file mode creation mask_.
+These permissions are set via the system api `umask` and are usually altered by the user with the `umask` command.
+In Unix shells, the `umask` command is built in to the shell, because the new value exists only for as long as the process that started it and any child process.
+Most systems also include `/usr/bin/umask` which is a stand-alone command which may be used to _retrieve_ the value of the mask.
+Using `/usr/bin/umask` can generate confusion as some may believe that setting the mask with `/usr/bin/umask` will persist this setting, but this is not the case as when the invocation of `/usr/bin/umask` concludes, the previous mask will be back in effect.
+For PowerShell customers, this becomes problematic as there is no easy way to invoke a persisted `umask` setting as there is no built-in command to change the `umask` of the PowerShell process.
+
+### Two new cmdlets
+
+This section of the RFC proposes two new cmdlets:
+
+* Get-Umask
+* Set-Umask
+
+These cmdlets (and associated engine support) will be available only on MacOS and Linux.
+`Get-Umask` retrieves the current process umask setting and `Set-Umask` sets the current process umask setting.
+For a good discussion of umask and its use, please visit: https://www.cyberciti.biz/tips/understanding-linux-unix-umask-value-usage.html.
+
+#### Get-Umask Syntax
+
+```powershell
+NAME
+    Get-Umask
+
+SYNTAX
+    Get-Umask [<CommonParameters>]
+
+ALIASES
+    None
+```
+
+#### Get-Umask Examples
+
+When running `Get-Umask` an object will be returned which contains the traditional umask string, the Symbolic representation of the mask, and an integer representation as follows:
+
+```powershell
+PS> get-umask
+
+Umask Symbolic        AsDecimal
```

Comment:
Glad to see this is getting tackled.

I don't think there's much value in including `AsDecimal`, because even the octal number is typically conceived of as an _array of digits_ rather than a single number.